Lib/test/test_tabnanny.py

```diff
@@ -0,0 +1,348 @@
+    def _validate_cmd(self, *args, return_code=0, stdout=b"", stderr=b""):
+        """Common function to assert the behaviour of command line interface.
+
+        Arguments:
+            * args        : Add arguments which will be directly fed as the
+                            arguments to tabnanny command line.
+            * return_code : Expected return code from tabnanny command line
+                            interface.
+            * stdout      : Expected standard output text after executing
+                            tabnanny command line interface.
+            * stderr      : Expected standard output error text after executing
+                            tabnanny command line interface.
+        """
+        rc, out, err = script_helper.assert_python_ok('-m', 'tabnanny', *args)
+        # Note: The `splitlines()` will solve the problem of CRLF(\r) added
+        # by OS Windows.
+        self.assertListEqual(out.splitlines(), stdout.splitlines())
+        self.assertEqual(rc, return_code)
+        self.assertListEqual(err.splitlines(), stderr.splitlines())
+
+    def test_with_errored_file(self):
+        """Should displays error when errored python file is given."""
+        file_path = create_tmp_python_file(SOURCE_CODES["wrong_indented"])
+        stderr = b"%r: Indentation Error: " % (file_path,)
+        stderr += (b'unindent does not match any outer indentation level'
+                   b' (<tokenize>, line 3)')
+        self._validate_cmd(file_path, stderr=stderr)
+
+    def test_with_error_free_file(self):
+        """Should not display anything if python file is correctly indented."""
+        file_path = create_tmp_python_file(SOURCE_CODES["error_free"])
+        self._validate_cmd(file_path)
+
+    def test_command_usage(self):
+        """Should display usage on no arguments."""
+        path = findfile('tabnanny.py')
+        stderr = b'Usage: %s [-v] file_or_directory ...' % (path.encode(),)
+        self._validate_cmd(stderr=stderr)
+
+    def test_quiet_flag(self):
+        """Should display less when quite mode is on."""
+        file_path = create_tmp_python_file(SOURCE_CODES["nannynag_errored"])
+        stdout = b"%s\n" % (file_path.encode('ascii'),)
+        self._validate_cmd("-q", file_path, stdout=stdout)
+
+    def test_verbose_mode(self):
+        """Should display more error information if verbose mode is on."""
+        file_path = create_tmp_python_file(SOURCE_CODES["nannynag_errored"])
+        stdout = b"%r: *** Line 3: trouble in tab city! ***\n" % (file_path,)
+        stdout += b"offending line: '\\tprint(\"world\")\\n'\n"
+        stdout += b"indent not equal e.g. at tab size 1\n"
+        self._validate_cmd("-v", file_path, stdout=stdout)
+
+    def test_double_verbose_mode(self):
+        """Should display detailed error information if double verbose is on."""
+        file_path = create_tmp_python_file(SOURCE_CODES["nannynag_errored"])
+        stdout = b"checking %r ...\n" % (file_path,)
```

Comment:
For better readability, you can use textwrap.dedent() and a multiline """...""" indented string. See test_faulthandler for examples.